crates/wasmi/src/engine/stack/frames.rs

```diff
@@ -72,20 +71,13 @@ impl CallStack {
         FuncFrame::new(ip, instance)
     }
 
-    /// Pushes a Wasm function onto the [`CallStack`].
-    pub(crate) fn push(
-        &mut self,
-        caller: &mut FuncFrame,
-        ip: InstructionPtr,
-        instance: Instance,
-    ) -> Result<FuncFrame, TrapCode> {
-        if self.len() == self.recursion_limit {
+    /// Pushes a Wasm caller function onto the [`CallStack`].
+    pub(crate) fn push(&mut self, caller: FuncFrame) -> Result<(), TrapCode> {
+        if self.len() >= self.recursion_limit {
```

Comment:
```suggestion
        if self.len() == self.recursion_limit {
```

Reply:
Please revert this. The `>=` is less precise with respect to the implied invariants since this number cannot possibly be larger than `self.recursion_limit` given that the code is bug free. So testing for `>=` in this position no longer properly communicated this invariant. Also it may not uncover possible bugs.

Reply:
Ok